src/hyperion/device_setup_plans/check_topup.py

```diff
@@ -0,0 +1,72 @@
+import bluesky.plan_stubs as bps
+from dodal.devices.detector import DetectorParams
+from dodal.devices.synchrotron import Synchrotron, SynchrotronMode
+
+from hyperion.log import LOGGER
+
+ALLOWED_MODES = [SynchrotronMode.USER, SynchrotronMode.SPECIAL]
+
+
+def _in_decay_mode(time_to_topup):
+    if time_to_topup == -1:
+        LOGGER.info("Machine in decay mode, gating disabled")
+        return True
+    return False
+
+
+def _gating_permitted(machine_mode):
+    if machine_mode not in ALLOWED_MODES:
+        LOGGER.info("Machine not in allowed mode, gating top up enabled.")
+        return True
+    return False
+
+
+def _delay_to_avoid_topup(total_exposure_time, time_to_topup):
+    if total_exposure_time > time_to_topup:
+        LOGGER.info(
+            """
+            Total exposure time + time needed for x ray centering exceeds time to
+            next top up. Collection delayed until top up done.
+            """
+        )
+        return True
+    LOGGER.info(
+        """
+        Total exposure time less than time to next topup. Proceeding with collection.
+        """
+    )
+    return False
+
+
+def check_topup_and_wait_if_necessary(
+    synchrotron: Synchrotron,
+    params: DetectorParams,
+    ops_time: float = 30.0,  # Account for xray centering, rotation speed, etc
```

Comment:
I'm not sure it makes sense to have a default at *this* level - my impression is that this plan should be generic across any collection type, and so it doesn't seem right to have a default here that *assumes* it's a particular type of collection we're running?